### PR TITLE
[tests-only] Wait for anyAfterLoading in file list views in tests

### PR DESCRIPTION
**Pull request by kulmann**
## Description
This PR gets rid of the navigation helper and instead waits for the `anyAfterLoading` selector of the filesList being present as a confirmation for the view being loaded. Previously we were waiting for the loading indicator component to appear (just waiting 1 second) and then to disappear again. This was basically sometimes skipped in tests, when the loading indicator was not visible on the screen, yet, when the respective navigation step in a scenario was reached. If it didn't appear within one second, the waiting for a loaded state would essentially be skipped. Since all our files list views have a structure where we know that it's having either a `not found`, `empty` or an actual files list, we can rely on waiting for that.

## Motivation and Context
Having more stable tests.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIRename</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14110/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPublic</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14110/

<p>







</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Files</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14110/

<p>







</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPublic</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14114/

<p>







</p></details>

**Comment by kulmann** [comment]
@individual-it what do you think about this PR? There are a few failing scenarios. All of them because a public link page with a password (obviously) doesn't have the elements from the `anyAfterLoading` selector available, but instead shows the password page for the public link. I'll need to change that. But other than that it seems to make the tests more stable. I noticed a lot of scenarios recently, when running tests locally, which showed that they failed to see `#files-list-progress` and skipped it then. I made this PR because I noticed a recent test failure in a different PR which showed the loading spinner (`#files-list-progress`) in the screenshots of failed tests, meaning that it showed after the timeout of 1 second.

**Comment by individual-it** [comment]
@kulmann if that is a reliably indicator for the page to be loaded, then go for it

**Comment by kulmann** [comment]
Rebased to current master and adjusted one test scenario. CI is green now.

File: tests/acceptance/features/webUISharingPublic/sharingPublicSession.feature [hunk 1]
```diff
@@ -17,17 +17,17 @@ Feature: Session storage for public link
 
   Scenario Outline: Public accesses the public link files page on a new session after visiting once
     Given user "Alice" has shared folder "<share-path>" with link with "read" permissions and password "pass123"
-    When the public tries to open the public link page of the last public link created by user "Alice" with password "pass123"
-    And the public opens the public link page of the last public link created by user "Alice" on a new session
-    Then the password input for the public link should appear in the webUI
-    When the user accesses the public link with password "pass123" using the webUI
+    When the public uses the webUI to access the last public link created by user "Alice" with password "pass123"
```
Comment (individual-it) [inline tests/acceptance/features/webUISharingPublic/sharingPublicSession.feature:20]: why changing this test?
I believe this test was written to start a complete new session of the browser (deleted cookies etc.)

Comment (kulmann) [reply tests/acceptance/features/webUISharingPublic/sharingPublicSession.feature:20]: Oh, true. I just looked for a step which handles the password correct and somehow ignored that it should be on a new session... will fix that, thanks!

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPublic</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14364/

<p>









</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Move</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14417/

<p>








</p></details>

**Comment by kulmann** [comment]
Rebased to current master. Let's see what CI does now...

File: tests/acceptance/stepDefinitions/publicLinkContext.js [hunk 2]
```diff
@@ -85,27 +68,32 @@ When(
...
-  'the public uses the webUI to access the last public link created by user {string} with password {string}',
-  async function(linkCreator, password) {
-    const lastShare = await sharingHelper.fetchLastPublicLinkShare(linkCreator)
-    if (lastShare.permissions === sharingHelper.PERMISSION_TYPES.create) {
-      await client.page.filesDropPage().navigateAndWaitForPasswordPage(lastShare.token)
-    } else {
-      await client.page.publicLinkFilesPage().navigateAndWaitForPasswordPage(lastShare.token)
-    }
-    return client.page.publicLinkPasswordPage().submitPublicLinkPassword(password)
+  'the public uses the webUI to access the last public link created by user {string} with password {string} on a new session',
```
Comment (individual-it) [inline tests/acceptance/stepDefinitions/publicLinkContext.js:78]: usually we place "using the webUI" at the end of the sentence, but that sound also fine for me @phil-davis comments?

**Review by individual-it — APPROVED** [review]

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPublicManagement</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14733/

<p>




</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIFiles2</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14764/

<p>










</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIFiles2</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14766/

<p>









</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIRename</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14766/

<p>









</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIFiles2</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14767/

<p>









</p></details>

**Pull request merged**